src/api_server/src/request/mmds.rs

```diff
@@ -3,37 +3,49 @@
 
 use crate::parsed_request::{Error, ParsedRequest};
 use crate::request::Body;
+use logger::{IncMetric, METRICS};
 use micro_http::StatusCode;
 use vmm::rpc_interface::VmmAction::SetMmdsConfiguration;
-use vmm::vmm_config::mmds::MmdsConfig;
 
 pub(crate) fn parse_get_mmds() -> Result<ParsedRequest, Error> {
+    METRICS.get_api_requests.mmds_count.inc();
     Ok(ParsedRequest::GetMMDS)
 }
 
 pub(crate) fn parse_put_mmds(
     body: &Body,
     path_second_token: Option<&&str>,
 ) -> Result<ParsedRequest, Error> {
+    METRICS.put_api_requests.mmds_count.inc();
     match path_second_token {
-        Some(config_path) => match *config_path {
-            "config" => Ok(ParsedRequest::new_sync(SetMmdsConfiguration(
-                serde_json::from_slice::<MmdsConfig>(body.raw()).map_err(Error::SerdeJson)?,
-            ))),
-            _ => Err(Error::Generic(
-                StatusCode::BadRequest,
-                format!("Unrecognized PUT request path `{}`.", *config_path),
-            )),
-        },
         None => Ok(ParsedRequest::PutMMDS(
-            serde_json::from_slice(body.raw()).map_err(Error::SerdeJson)?,
+            serde_json::from_slice(body.raw()).map_err(|e| {
+                METRICS.put_api_requests.mmds_fails.inc();
+                Error::SerdeJson(e)
+            })?,
         )),
+        Some(&"config") => Ok(ParsedRequest::new_sync(SetMmdsConfiguration(
+            serde_json::from_slice(body.raw()).map_err(|e| {
+                METRICS.put_api_requests.mmds_fails.inc();
+                Error::SerdeJson(e)
+            })?,
+        ))),
+        Some(&unrecognized) => {
+            METRICS.put_api_requests.mmds_fails.inc();
+            Err(Error::Generic(
+                StatusCode::BadRequest,
+                format!("Unrecognized PUT request path `{}`.", unrecognized),
+            ))
+        }
     }
 }
 
 pub(crate) fn parse_patch_mmds(body: &Body) -> Result<ParsedRequest, Error> {
     Ok(ParsedRequest::PatchMMDS(
```

Comment:
Missing `METRICS.patch_api_requests.mmds_count.inc();` here

Reply:
OMG :facepalm: . Sorry for that. Fixed. Tests adjusted as well.